### PR TITLE
add sigmoid cross entropy with logits to kl2

### PR DESCRIPTION
**Pull request by houj04**
### PR types
New features

### PR changes
OPs

### Describe
sigmoid_cross_entropy_with_logits_op_xpu: support kl2. support 'ignore_index' and 'normalize'.

### CODE REVIEW

**Review by chenwhql — DISMISSED** [review]
LGTM for PADDLE_ENFORCE

File: paddle/fluid/operators/sigmoid_cross_entropy_with_logits_op_xpu.cc [hunk 1]
```diff
@@ -41,24 +41,46 @@ class SigmoidCrossEntropyWithLogitsXPUKernel : public framework::OpKernel<T> {
...
-        platform::errors::InvalidArgument("normalize only support true now."));
     int ignore_index = context.Attr<int>("ignore_index");
-    PADDLE_ENFORCE_EQ(ignore_index, kIgnoreIndex,
-                      platform::errors::InvalidArgument(
-                          "ignore_index only support %d now.", kIgnoreIndex));
+    bool normalize = context.Attr<bool>("normalize");
+
+    // allocate temp memory
+    xpu::ctx_guard RAII_GUARD(dev_ctx.x_context());
+    int* hit = RAII_GUARD.alloc_l3_or_gm<int>(input->numel());
```
Comment (QingshuChen) [inline paddle/fluid/operators/sigmoid_cross_entropy_with_logits_op_xpu.cc:50]: 判断下hit的返回值是不是为null

File: paddle/fluid/operators/sigmoid_cross_entropy_with_logits_op_xpu.cc [hunk 2]
```diff
@@ -41,24 +41,46 @@ class SigmoidCrossEntropyWithLogitsXPUKernel : public framework::OpKernel<T> {
...
         r, XPU_SUCCESS,
         platform::errors::External("XPU sigmoid_cross_entropy_with_logits "
                                    "kernel return wrong value[%d %s]",
                                    r, XPUAPIErrorMsg[r]));
+    if (normalize) {
+      int* non_zero = RAII_GUARD.alloc_l3_or_gm<int>(1);
+      int r = xpu::nonzero_count(dev_ctx.x_context(),
+                                 reinterpret_cast<const XPUType*>(hit),
+                                 non_zero, input->numel());
+      PADDLE_ENFORCE_EQ(r, XPU_SUCCESS, platform::errors::External(
```
Comment (QingshuChen) [inline paddle/fluid/operators/sigmoid_cross_entropy_with_logits_op_xpu.cc]: 有一个XDNN的宏，用那个判断返回值

File: paddle/fluid/operators/sigmoid_cross_entropy_with_logits_op_xpu.cc (hunk 1 above)
Comment (houj04) [reply paddle/fluid/operators/sigmoid_cross_entropy_with_logits_op_xpu.cc:50]: done

File: paddle/fluid/operators/sigmoid_cross_entropy_with_logits_op_xpu.cc (hunk 2 above)
Comment (houj04) [reply paddle/fluid/operators/sigmoid_cross_entropy_with_logits_op_xpu.cc]: done

**Review by HarperCy — APPROVED** [review]

**Review by QingshuChen — APPROVED** [review]
LGTM
